irc/nickserv.go

```diff
@@ -194,12 +186,257 @@ password by supplying their username and then the desired password.`,
+	// without creating a compile-time initialization loop
+	rb.Add(nil, "NickServ!NickServ@localhost", "NOTICE", rb.target.Nick(), text)
+}
+
+func nsGetHandler(server *Server, client *Client, command string, params []string, rb *ResponseBuffer) {
+	var account string
+	if command == "saget" {
+		account = params[0]
+		params = params[1:]
+	} else {
+		account = client.Account()
+	}
+
+	accountData, err := server.accounts.LoadAccount(account)
+	if err == errAccountDoesNotExist {
+		nsNotice(rb, client.t("No such account"))
+		return
+	} else if err != nil {
+		nsNotice(rb, client.t("Error loading account data"))
+		return
+	}
+
+	displaySetting(params[0], accountData.Settings, client, rb)
+}
+
+func displaySetting(settingName string, settings AccountSettings, client *Client, rb *ResponseBuffer) {
+	config := client.server.Config()
+	switch strings.ToLower(settingName) {
+	case "enforce":
+		storedValue := settings.NickEnforcement
+		serializedStoredValue := nickReservationToString(storedValue)
+		nsNotice(rb, fmt.Sprintf(client.t("Your stored nickname enforcement setting is: %s"), serializedStoredValue))
+		serializedActualValue := nickReservationToString(configuredEnforcementMethod(config, storedValue))
+		nsNotice(rb, fmt.Sprintf(client.t("Given current server settings, your nickname is enforced with: %s"), serializedActualValue))
+	case "autoreplay-lines":
+		if settings.AutoreplayLines == nil {
+			nsNotice(rb, fmt.Sprintf(client.t("You will receive the server default of %d lines of autoreplayed history"), config.History.AutoreplayOnJoin))
+		} else {
+			nsNotice(rb, fmt.Sprintf(client.t("You will receive %d lines of autoreplayed history"), *settings.AutoreplayLines))
+		}
+	case "autoreplay-joins":
+		if settings.AutoreplayJoins {
+			nsNotice(rb, client.t("You will see JOINs and PARTs in autoreplayed history lines"))
+		} else {
+			nsNotice(rb, client.t("You will not see JOINs and PARTs in autoreplayed history lines"))
+		}
+	case "bouncer":
+		if !config.Accounts.Bouncer.Enabled {
+			nsNotice(rb, fmt.Sprintf(client.t("This feature has been disabled by the server administrators")))
+		} else {
+			switch settings.AllowBouncer {
+			case BouncerAllowedServerDefault:
+				if config.Accounts.Bouncer.AllowedByDefault {
+					nsNotice(rb, fmt.Sprintf(client.t("Bouncer functionality is currently enabled for your account, but you can opt out")))
+				} else {
+					nsNotice(rb, fmt.Sprintf(client.t("Bouncer functionality is currently disabled for your account, but you can opt in")))
+				}
+			case BouncerDisallowedByUser:
+				nsNotice(rb, fmt.Sprintf(client.t("Bouncer functionality is currently disabled for your account")))
+			case BouncerAllowedByUser:
+				nsNotice(rb, fmt.Sprintf(client.t("Bouncer functionality is currently enabled for your account")))
+			}
+		}
+	default:
+		nsNotice(rb, client.t("No such setting"))
+	}
+}
+
+func stringToBool(str string) (result bool, err error) {
```

Comment:
This function should go into utils, and be something more along the lines of these options/etc:

```
str[0] == 't', 'y', '1': true
str == 'on': true
str[0] == 'f', 'n', '0': false
str == 'off': false
```